torch/export/export_schema.py

```diff
@@ -0,0 +1,326 @@
+    #   // - "length" (optional) - number of bytes containing data. Integer stored as string.
+    #   // - "checksum" (optional) - SHA1 digest of file specified in under 'location' key.
+    #   repeated StringStringEntryProto external_data
+    location: str
+    offset: str     # !!! Consider using int, but int has int_max limitation
+    length: str     # !!! Consider using int, but int has int_max limitation
+    checksum: str
+
+
+@dataclass
+class Storage:
+    class DataLocation(Enum):
+        Default = auto()
+        External = auto()
+
+    data_location: DataLocation
+    data: Union[Buffer, ExternalBuffer]
+
+
+# This is a concrete tensor backed by storage
+@dataclass
+class Tensor:
+    # storage
+    storage: Storage
+
+    # metadata
+    meta: TensorMeta
+
+
+# !!! consider adding an optional named field to Tensor directly to avoid defining a new class
+@dataclass
+class NamedTesnsor:
+    name: str
+    tensor: Tensor
+
+################################################################################
+# Following section is the defining the schema of 3 level construct: GraphModule, Graph, Node
+
+# IValue has no corresponding class in fx
+# IValue is the "values" that are passed between nodes in the graph
+# !!! Assumption: Only Tensor can be passed between nodes, and not other int/float/bool types...
+# IValue is a named virtual tensor, with an optional TensorMeta that describes the properties of the tensor
+# !!! Consider using a more descriptive name, e.g. TensorValue, TensorPlaceholder, TensorArgument, etc.
+@dataclass
+class IValue:
+    name: str   # A unique identifider name for the IValue
+                # The name will be used in the graph and node to refer to the IValue
+
+    meta: TensorMeta
+
+
+# Maps to fx.Node
+@dataclass
+class Node:
+    op: str         # One of ['placeholder', 'call_function', 'get_attr', 'output'],
+                    # call_method and call_module are not supported, as they shouldn't apprear in the Caononical FX Graph
+                    # !!! Consider using an enum instead of string
+
+    target: str      # fully qualified name to the target, e.g. aten.add.Tensnor
+                     # !!! Consider using a structured operator name instead of string
+
+    args: List[Argument]              # args for this node
+    kwargs: List[KeywordArgument]     # kwargs for this node
+                                      # !!! Not all types in Argument are used as kwargs, e.g. TensorArgument should not be used as kwargs
+                                      # Do we want to enforce this in the schema? i.e. only allow certain types to be used as kwargs?
+
+    outputs: List[TensorArgument]   # A list of Tensor Argument returned by this node
+                                    # !!! Notice: this assumes that a node can only return Tensor(s), and not other int/float/bool types...
+
+    metadata: List[StringStringPair]
```

Comment:
We should just spell out metadata as we dont allow arbitrary metadata. 

i.e. metadata: Metadata
Then Metadata is a dataclass with fixed fields: such as val, stack_trace etc.

Reply:
+1 I think at least we should specify what *must* be in here (e.g. source info).

Reply:
OK. We can have Metadata dataclass with fixed fields, but I think we also need to have a field for flexible string-string map for extensibility. 

Otherwise, we will need to modify the schema every time a new meta field is needed.

Reply:
> Otherwise, we will need to modify the schema every time a new meta field is needed.

Seems like a good thing? If the metadata affects the functionality/correctness of the serialized format, shouldn't we be required to schematize it and maintain BC/FC for it?

Reply:
Do we need this for extensibility? Is it at all possible to make `export` emit additional custom metadata? Otherwise we should lock it down and explicitly version on changes.

Reply:
We need this for extensibility. It's very likely that we will get request asking to add custom metadata. Actually, I just got one request today to add the source torch api call to the metadata.